cytotable/convert.py

```diff
@@ -199,21 +192,27 @@ def _get_table_chunk_offsets(
             file or table of some kind.
         chunk_size: int
             The size in rowcount of the chunks to create.
+        page_key: str
+            The column name to be used as the key for pagination.
+            Expected to be of numeric type (int, float) for ordering.
+        sql_stmt:
+            Optional sql statement to form the pagination set from.
+            Default behavior extracts pagination sets from the full
+            data source.
 
     Returns:
-        List[int]
-            List of integers which represent offsets to use for reading
-            the data later on.
+        List[Any]
+            List of keys to use for reading the data later on.
     """
 
     import logging
-    import pathlib
+    import sqlite3
+    from contextlib import closing
```

Comment:
Why do imports happen at two points in this file? I could be mistaken, but isn't it more standard to have all imports at the start of the file?

Reply:
That's generally a good point, but I believe this is because the parallelization library that @d33bs is using, [parsl](https://parsl-project.org/), requires that functions that are parallelized import all the libraries they need in the function, not within the enclosing module like you'd normally do. You can tell it's an entrypoint from the `@python_app` decorator before the function definition. 

Here's a bit from the parsl docs at least saying it's necessary to import within parallelized functions: https://parsl.readthedocs.io/en/stable/1-parsl-introduction.html#Python-Apps.

Reply:
Thanks Jenna for a good question and Faisal for the docs. Concurring with what Faisal mentioned, this is to help with multiprocessing through Parsl where each decorated function might not have access of the full module's imports.